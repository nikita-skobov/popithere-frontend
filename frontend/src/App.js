import React, { Component } from 'react'
import PropTypes from 'prop-types'

const tester = () => {
  const test = 'yupp'
  return {
    hello: 'wolrd',
    suze: 'yes',
    bob: 100,
    test,
  }
}

class App extends Component {
  constructor(props) {
    super(props)
    this.name = props.name
  }

  render() {
    const {
      hello,
      suze,
      bob,
      test,
    } = tester()

    return (
      <div className="App">
        <h2>hello {this.name}</h2>
        <h3>hello {hello}</h3>
        <h4>hello {suze}</h4>
        <h5>hello {bob}</h5>
        <h6>hello {test}</h6>
      </div>
    )
  }
}

App.defaultProps = {
  name: null,
}

App.propTypes = {
  name: PropTypes.string,
}

export default App
