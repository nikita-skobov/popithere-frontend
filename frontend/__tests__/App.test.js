/* global test, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import App from '../src/App'

test('App works', () => {
  const component = renderer.create(
    <App name="myname2" />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
