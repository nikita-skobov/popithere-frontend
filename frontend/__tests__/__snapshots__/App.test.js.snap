// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App works 1`] = `
<div
  className="App"
>
  <h2>
    hello 
    myname2
  </h2>
  <h3>
    hello 
    wolrd
  </h3>
  <h4>
    hello 
    yes
  </h4>
  <h5>
    hello 
    100
  </h5>
  <h6>
    hello 
    yupp
  </h6>
</div>
`;
